src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] array;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        array = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (++size > threshold) {
+            array = resize();
+        }
```

Comment:
It's better to reassign the array field in the resize() method so that single responsibility principle is not violated

Reply:
Fixed but what is the "principle is not violated"?